tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeRange.java

```diff
@@ -0,0 +1,279 @@
+   */
+  public boolean contains(long value) {
+    return min <= value && max >= value;
+  }
+
+  /**
+   * @return true if the given range lies in this range, inclusively
+   */
+  public boolean contains(TimeRange r) {
+    return min <= r.min && max >= r.max;
+  }
+
+  /**
+   * @param min
+   * @param max
+   */
+  public void set(long min, long max) {
+    this.min = min;
+    this.max = max;
+
+    sort();
+  }
+
+  /**
+   * @param r
+   */
+  public void set(TimeRange r) {
+    set(r.getMin(), r.getMax());
+  }
+
+  /**
+   * @return The lower range boundary
+   */
+  public long getMin() {
+    return min;
+  }
+
+  /**
+   * @param min
+   */
+  public void setMin(long min) {
+    this.min = min;
+
+    sort();
+  }
+
+  /**
+   * @return The upper range boundary
+   */
+  public long getMax() {
+    return max;
+  }
+
+  /**
+   * @param max
+   */
+  public void setMax(long max) {
+    this.max = max;
+
+    sort();
+  }
+
+  private void sort() {
+    if (min > max) {
+      long t = min;
+      min = max;
+      max = t;
+    }
+  }
+
+  /**
+   * @return <code>true</code> if the intersection exists
+   */
+  public boolean intersection(TimeRange r, TimeRange dest) {
+    if (intersects(r)) {
+      dest.set(Math.max(min, r.min), Math.min(max, r.max));
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * @return <code>true</code> if the ranges have values in common
+   */
+  public boolean intersects(TimeRange r) {
+    return overlaps(min, max, r.min, r.max);
+  }
+
+  /**
+   * @return <code>true</code> if the ranges overlap
+   */
+  public static boolean overlaps(long minA, long maxA, long minB, long maxB) {
+    assert minA <= maxA;
+    assert minB <= maxB;
+
+    // Because timestamp is long data type, x and x+1 are considered continuous.
+    return !(minA >= maxB + 2 || maxA <= minB - 2);
+  }
+
+  @Override
+  public String toString() {
+    return "[ " + min + " : " + max + " ]";
+  }
+
+  // NOTE the primitive timeRange is always a closed interval [min,max] and
+  // only in getRemains functions are leftClose and rightClose considered.
+  private boolean leftClose = true; // default true
+  private boolean rightClose = true; // default true
+
+  public void setLeftClose(boolean leftClose) {
+    this.leftClose = leftClose;
+  }
+
+  public void setRightClose(boolean rightClose) {
+    this.rightClose = rightClose;
+  }
+
+  public boolean getLeftClose() {
+    return leftClose;
+  }
+
+  public boolean getRightClose() {
+    return rightClose;
+  }
+
+  /**
+   * Get the remaining time ranges in the current ranges but not in timeRangesPrev.
+   *
+   * NOTE the primitive timeRange is always a closed interval [min,max] and only in this function
+   * are leftClose and rightClose changed.
+   *
+   * @param timeRangesPrev time ranges union in ascending order
+   * @return the remaining time ranges
+   */
+  public ArrayList<TimeRange> getRemains(ArrayList<TimeRange> timeRangesPrev) {
```

Comment:
return type of method usually use interface such as ``` List```  rather than implementation ```ArrayList```